src/PrestaShopBundle/Routing/Converter/RouterProvider.php

```diff
@@ -45,43 +49,48 @@ class RouterProvider extends AbstractLegacyRouteProvider
     private $router;
 
     /**
-     * RouterProvider constructor.
-     *
-     * @param RouterInterface $router
+     * @var FeatureFlagRepository
      */
-    public function __construct(RouterInterface $router)
+    private $featureFlagRepository;
+
+    public function __construct(RouterInterface $router, FeatureFlagRepository $featureFlagRepository)
     {
         $this->router = $router;
+        $this->featureFlagRepository = $featureFlagRepository;
     }
 
     /**
      * {@inheritdoc}
      */
     public function getLegacyRoutes()
     {
-        if (null == $this->legacyRoutes) {
-            $this->legacyRoutes = $this->buildLegacyRoutes();
+        if (null !== $this->legacyRoutes) {
+            return $this->legacyRoutes;
+        }
+
+        $this->legacyRoutes = [];
+        /** @var Route $route */
+        foreach ($this->router->getRouteCollection() as $routeName => $route) {
+            if ($this->isLegacyRoute($route)) {
+                $this->legacyRoutes[$routeName] = LegacyRoute::buildLegacyRoute($routeName, $route->getDefaults());
+            }
         }
 
         return $this->legacyRoutes;
     }
 
-    /**
-     * @return array
-     */
-    private function buildLegacyRoutes()
+    public function isLegacyRoute(Route $route): bool
```

Comment:
I don't think this needs to be public, it's only used internally The only need you have for it as public is for the tests
But the tests shouldn't force the implementation, you'll probably have to find some other way to test this I guess